### PR TITLE
[Rest Server] Update host list env variable

### PR DESCRIPTION
**Pull request by abuccts**
- Update host list env variable, convert `PAI_TASK_ROLE_$i_HOST_LIST` to
`PAI_TASK_ROLE_$name_HOST_LIST`
- Add `PAI_TASK_ROLE_LIST`

### CODE REVIEW

**Comment by fanyangCS** [comment]
> | `taskRole.name`                | String, required           | Name for the task role, need to be unique with other roles |

It seems your current implementation assumes "name" must not contain space. what is our assumption to the format of name? perhaps a regex to check the name when job submission. and clarify the format requirement here?

---
Refers to: job-tutorial/README.md:102 in cebfbb1. [](commit_id = cebfbb13f1965589c5851f2848fed6306f28c61d, deletion_comment = False)

**Comment by fanyangCS** [comment]
> hdfs dfs -touchz $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/$APP_ID-$PAI_TASK_ROLE_NO-$PAI_TASK_INDEX-$PAI_CURRENT_CONTAINER_IP-$PAI_CURRENT_CONTAINER_PORT || exit 1

this has been removed, right?

---
Refers to: rest-server/src/templates/dockerContainerScript.mustache:59 in cebfbb1. [](commit_id = cebfbb13f1965589c5851f2848fed6306f28c61d, deletion_comment = False)

**Comment by fanyangCS** [comment]
> export PAI_TASK_ROLE_NO={{{ idx }}}

this has been removed. right?

---
Refers to: rest-server/src/templates/dockerContainerScript.mustache:51 in cebfbb1. [](commit_id = cebfbb13f1965589c5851f2848fed6306f28c61d, deletion_comment = False)

File: job-tutorial/README.md [hunk 1]
```diff
@@ -138,19 +138,19 @@ Below we show a complete list of environment variables accessible in a Docker co
 | PAI_DATA_DIR                       | `dataDir` in config file                 |
 | PAI_OUTPUT_DIR                     | `outputDir`in config file or the generated path if `outputDir` is not specified |
 | PAI_CODE_DIR                       | `codeDir` in config file                 |
+| PAI_TASK_ROLE_LIST                 | Comma separated all task role names in config file |
 | PAI_TASK_ROLE_NAME                 | `taskRole.name` of current task role     |
 | PAI_TASK_ROLE_NUM                  | `taskRole.number` of current task role   |
```
Comment (fanyangCS) [inline job-tutorial/README.md]: >taskRole.number [](start = 40, length = 15)

there is no taskRole.number, do you mean `taskRole.taskNumber` or the number of taskRole?

Comment (abuccts) [reply job-tutorial/README.md]: yes, updated

**Comment by abuccts** [comment]
Do we need to remove `PAI_TASK_ROLE_NO` from code? I have already removed it in the document.

File: job-tutorial/README.md [hunk 2]
```diff
@@ -138,19 +138,19 @@ Below we show a complete list of environment variables accessible in a Docker co
 | PAI_DATA_DIR                       | `dataDir` in config file                 |
 | PAI_OUTPUT_DIR                     | `outputDir`in config file or the generated path if `outputDir` is not specified |
 | PAI_CODE_DIR                       | `codeDir` in config file                 |
+| PAI_TASK_ROLE_LIST                 | Comma separated all task role names in config file |
 | PAI_TASK_ROLE_NAME                 | `taskRole.name` of current task role     |
-| PAI_TASK_ROLE_NUM                  | `taskRole.number` of current task role   |
+| PAI_TASK_ROLE_NUM                  | `taskRole.taskNumber` of current task role |
```
Comment (sterowang) [inline job-tutorial/README.md]: It seemed task role num mean number of task roles. Shall you gives out a better name without this kind of confusion?

File: rest-server/src/templates/dockerContainerScript.mustache [hunk 3]
```diff
@@ -42,6 +42,7 @@ export PAI_USERNAME={{{ jobData.username }}}
 export PAI_DATA_DIR={{{ jobData.dataDir }}}
 export PAI_OUTPUT_DIR={{{ jobData.outputDir }}}
 export PAI_CODE_DIR={{{ jobData.codeDir }}}
+export PAI_TASK_ROLE_LIST={{{ taskRoleList }}}
 export PAI_TASK_ROLE_NAME={{{ taskData.name }}}
 export PAI_TASK_ROLE_NUM={{{ taskData.taskNumber }}}
```
Comment (sterowang) [inline rest-server/src/templates/dockerContainerScript.mustache]: See comments above, name looks a little ambiguous.

File: job-tutorial/README.md (hunk 2 above)
Comment (sterowang) [reply job-tutorial/README.md]: Maybe PAI_TASK_NUM is enough?

File: job-tutorial/README.md [hunk 4]
```diff
@@ -138,19 +138,19 @@ Below we show a complete list of environment variables accessible in a Docker co
...
 | PAI_TASK_CPU_NUM                   | `taskRole.cpuNumber` of current task     |
 | PAI_TASK_MEM_MB                    | `taskRole.memoryMB` of current task      |
 | PAI_TASK_GPU_NUM                   | `taskRole.gpuNumber` of current task     |
 | PAI_TASK_ROLE_INDEX                | Index of current task in the task role, starting from 0 |
```
Comment (sterowang) [inline job-tutorial/README.md]: PAI_CURRENT_TASK_INDEX

**Comment by fanyangCS** [comment]
yes, please remove it from code.

---
In reply to: [364913937](https://github.com/Microsoft/pai/pull/322#issuecomment-364913937) [](ancestors = 364913937)

**Comment by abuccts** [comment]
removed

File: job-tutorial/README.md (hunk 2 above)
Comment (abuccts) [reply job-tutorial/README.md]: ok

**Pull request merged**